e2e/cypress/integration/integrations/custom_slash_commands/slash_commands_spec.js

```diff
@@ -0,0 +1,213 @@
+    });
+
+    it('MM-T694 Error: trigger word in use', () => {
+        // # Add new command
+        cy.get('#addSlashCommand').click();
+
+        // # Type a trigger word and URL
+        cy.get('#trigger').type('my_trigger_word');
+        cy.get('#url').type('http://');
+
+        // # Save
+        cy.get('#saveCommand').click();
+
+        // # Go to integrations
+        cy.visit(`/${testTeam.name}/integrations`);
+
+        // * Validate that slash command section is enabled
+        cy.get('#slashCommands').should('be.visible');
+
+        // # Open slash command page
+        cy.get('#slashCommands').click();
+
+        // # Add same command
+        cy.get('#addSlashCommand').click();
+
+        // # Type same trigger word and URL
+        cy.get('#trigger').type('my_trigger_word');
+        cy.get('#url').type('http://');
+
+        // # Save
+        cy.get('#saveCommand').click();
+
+        // * Verify that we get the error message
+        cy.findByText('This trigger word is already in use. Please choose another word.').should('exist').and('be.visible').scrollIntoView();
+    });
+
+    it('MM-T695 Run custom slash command', () => {
+        addNewCommand(trigger);
+        runSlashCommand(testTeam, trigger);
+    });
+
+    it('MM-T698 Cancel out of edit', () => {
+        addNewCommand(trigger);
+
+        // # Go to integrations
+        cy.visit(`/${testTeam.name}/integrations`);
+
+        // # Open slash command page
+        cy.get('#slashCommands').click();
+
+        // # Click on edit
+        cy.get('a[href*="/edit"]').click();
+
+        // # Change url
+        cy.get('#url').clear().type('http://mattermost.com');
+
+        // # Click on Cancel
+        cy.get('a').contains('Cancel').click();
+
+        // # Click on edit again
+        cy.get('a[href*="/edit"]').click();
+
+        // * Verify that url value is not changed
+        cy.get('#url').should('have.value', 'http://hidden-peak-21733.herokuapp.com/test_inchannel');
+    });
+
+    it('MM-T699 Edit custom slash command', () => {
+        addNewCommand(trigger);
+
+        // # Go to integrations
+        cy.visit(`/${testTeam.name}/integrations`);
+
+        // # Open slash command page
+        cy.get('#slashCommands').click();
+
+        // # Click on edit
+        cy.get('a[href*="/edit"]').click();
+
+        // # Update display name
+        cy.get('#displayName').clear().type('Test Message - Edit');
+
+        // # Update
+        cy.get('#saveCommand').click();
+
+        // * Verify successful update
+        cy.findByText('Test Message - Edit').should('exist').and('be.visible');
+
+        runSlashCommand(testTeam, trigger);
+    });
+});
+
+function addNewCommand(trigger) {
+    // # Add new command
+    cy.get('#addSlashCommand').click();
+
+    // # Type a trigger word, url and display name
+    cy.get('#trigger').type(`${trigger}`);
+    cy.get('#url').type('http://hidden-peak-21733.herokuapp.com/test_inchannel');
```

Comment:
I didn't know that there's a live URL like this but I suggest not to use it for Cypress testing. We usually spin a local node server for webhook testing by `npm run start:webhook`.

Then, url could be generated by,
```javascript
cy.apiGetChannelByName(testTeam.name, 'town-square').then((res) => {
    const url = `${Cypress.env('webhookBaseUrl')}/send_message_to_channel?channel_id=${res.body.id}`;
});
```

Be sure to add verification in before hook so it'll fail immediately when no node server is running/accessible.
```javascript
before(() => {
    cy.requireWebhookServer();
});
```